File: benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/LatencyOperation.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.Benchmark.Mode;
import de.hhu.bsinfo.observatory.benchmark.result.LatencyMeasurement;

public abstract class LatencyOperation extends Operation {

    LatencyOperation(Benchmark benchmark, Mode mode, int operationCount, int operationSize) {
        super(benchmark, mode, new LatencyMeasurement(operationCount, operationSize));
    }

    @Override
    protected LatencyMeasurement getMeasurement() {
        return (LatencyMeasurement) super.getMeasurement();
    }
}

File: cpp/src/observatory/operation/LatencyOperation.cpp
#include "LatencyOperation.h"

namespace Observatory {

LatencyOperation::LatencyOperation(Benchmark *benchmark, Benchmark::Mode mode, uint32_t operationCount, uint32_t operationSize) :
        Operation(benchmark, mode, std::make_shared<LatencyMeasurement>(operationCount, operationSize)) {

}

LatencyMeasurement& LatencyOperation::getMeasurement() const {
    return dynamic_cast<LatencyMeasurement&>(Operation::getMeasurement());
}

const char *LatencyOperation::getClassName() const {
    return "LatencyOperation";
}

}